light_sdk/src/reading/mod.rs

```diff
@@ -0,0 +1,679 @@
+use std::str::FromStr;
```

Comment:
Overall I like the api.

It's probably better to split the individual submodules into their own files.